components/autocomplete/index.js

```diff
@@ -131,6 +131,14 @@ export class Autocomplete extends Component {
 		this.search = this.search.bind( this );
 		this.handleKeyDown = this.handleKeyDown.bind( this );
 		this.getWordRect = this.getWordRect.bind( this );
+		this.debouncedSearch = debounce( ( e ) => {
+			this.search( e );
+		}, 250 );
+		this.doSearch = ( event ) => {
+			// Ensure the synthetic event can be handled asynchronously.
+			event.persist();
+			this.debouncedSearch( event );
```

Comment:
aha! I struggled to figure this out, I thought because of React's synthetic events I had to `persist` when debouncing... So if I extract search hereI won't need the event later. I'll change that.